process/block/preprocess/basePreProcess.go

```diff
@@ -415,3 +415,21 @@ func (bpp *basePreProcess) getTxMaxTotalCost(txHandler data.TransactionHandler)
 
 	return cost
 }
+
+func (bpp *basePreProcess) getTotalGasConsumed() uint64 {
+	totalGasConsumed := uint64(0)
+	if bpp.gasHandler.TotalGasConsumed() > bpp.gasHandler.TotalGasRefunded() {
+		totalGasConsumed = bpp.gasHandler.TotalGasConsumed() - bpp.gasHandler.TotalGasRefunded()
+	}
+
+	return totalGasConsumed
```

Comment:
it is ok to return 0 here if the TotalGasConsumed <= TotalGasRefunded  ?

Reply:
This situation, to refund more than you received to be used, should never occur. But if this will happen because of an introduced bug, at least it is protected to against overflow situation, as both of them are uint64. So you consider that nothing has been processed yet.